packets.go

```diff
@@ -326,6 +368,21 @@ func (mc *mysqlConn) writeAuthPacket(cipher []byte) error {
 	// Assume native client during response
 	pos += copy(data[pos:], "mysql_native_password")
 	data[pos] = 0x00
+	pos++
+
+	// Connection attributes
+	if attrlen > 0 {
+		data[pos] = byte(attrlen)
+		pos++
+
+		for attrname, attrvalue := range attrs {
```

Comment:
@dveeden That what I meant. You could directly pack this into one byte sequence / string when the params are parsed.

But forget that, the Config is exported for a reason now.